File: src/test/java/javabot/operations/LiteralOperationTest.java
package javabot.operations;

import javabot.BaseMessagingTest;
import org.testng.annotations.Test;

import java.util.Date;

@Test(groups = "operations")
public class LiteralOperationTest extends BaseMessagingTest {
  @Test
  public void testMissingFactoid() {
    String factoidName = "foo" + new Date().getTime();
    testMessage("~literal " + factoidName,
        "I have no factoid called \"" + factoidName + "\"");
  }

  // TODO needs "existing factoid test"
}

File: src/test/kotlin/javabot/operations/LiteralOperationTest.kt
package javabot.operations

import com.antwerkz.sofia.Sofia
import jakarta.inject.Inject
import java.util.Date
import javabot.BaseTest
import org.testng.Assert
import org.testng.annotations.Test

@Test(groups = arrayOf("operations"))
class LiteralOperationTest : BaseTest() {
    @Inject private lateinit var operation: LiteralOperation

    @Test
    fun testMissingFactoid() {
        val factoidName = "foo${Date().time}"
        var response = operation.handleMessage(message("~literal ${factoidName}"))
        Assert.assertEquals(response[0].value, Sofia.factoidUnknown(factoidName))
    }

    // TODO needs "existing factoid test"
}
